samplers/samplers_test.go

```diff
@@ -22,32 +22,28 @@ func TestCounterEmpty(t *testing.T) {
 	assert.Len(t, metrics, 1, "Flushes 1 metric")
 
 	m1 := metrics[0]
-	assert.Equal(t, int32(10), m1.Interval, "Interval")
-	assert.Equal(t, "rate", m1.MetricType, "Type")
+	assert.Equal(t, CounterMetric, m1.Type, "Type")
 	assert.Len(t, c.Tags, 1, "Tag length")
 	assert.Equal(t, c.Tags[0], "a:b", "Tag contents")
-	// The counter returns an array with a single tuple of timestamp,value
-	assert.Equal(t, 0.1, m1.Value[0][1], "Metric value")
+	assert.Equal(t, float64(1), m1.Value, "Metric value")
 }
 
 func TestCounterRate(t *testing.T) {
 	c := NewCounter("a.b.c", []string{"a:b"})
 
 	c.Sample(5, 1.0)
 
-	// The counter returns an array with a single tuple of timestamp,value
 	metrics := c.Flush(10 * time.Second)
-	assert.Equal(t, 0.5, metrics[0].Value[0][1], "Metric value")
+	assert.Equal(t, float64(5), metrics[0].Value, "Metric value")
```

Comment:
So to be clear: the reason this value has changed is that we're now letting the sampler provide the "true" value, and the per-interval rate calculation is being done within the plugin (ie, by Datadog)?